tests/testing/assertions_test.py

```diff
@@ -614,31 +614,48 @@ def test_assert_similar_binned_data_array_bin_coord_mismatch(
 @pytest.mark.parametrize(
     'a',
     (
-        sc.array(dims=['x'], values=np.clip(np.random.randn(10), -10, 10)),
+        sc.array(dims=['x'], values=[1000, -1, 0.001, -2.3]),
         sc.scalar(2.3),
         sc.scalar(-2.0, unit='m'),
+        sc.DataArray(
+            sc.ones(sizes={'rv': 9}),
+            coords={'rv': sc.arange('rv', 9), 'j': sc.scalar(3.1)},
+        ),
     ),
 )
 @pytest.mark.parametrize('rtol', (1e-8, 1e-5, 1e-3, 1e-1))
 def test_allclose_different_tolerances(rtol, a):
-    if hasattr(a, 'unit'):
-        rtol = sc.scalar(rtol, unit='dimensionless')
-
-    if isinstance(a, sc.Variable):
-        b = a.copy()
-        b.values += (
-            0.9 * rtol.value * a.values * (2 * np.random.random(a.values.shape) - 1)
-        )
-    else:
-        b = a + 0.9 * rtol * a * (2 * np.random.random() - 1)
-
-    print(a, b)
+    rtol = sc.scalar(rtol, unit='dimensionless')
+    b = a.copy()
+    av = a.values if isinstance(a, sc.Variable) else a.data.values
+    bv = b.values if isinstance(b, sc.Variable) else b.data.values
+    bv += 0.9 * rtol.value * av * (2 * np.random.random(bv.shape) - 1)
     assert_allclose(a, b, rtol=rtol)
+    bv += rtol.value * np.sign(bv - av) * np.abs(bv)
```

Comment:
What is going on here? This seems like a pretty complicated way of handling tolerances. I'd prefer to keep logic (esp floating point related logic) out of tests. So, here, I'd duplicate the test and manually adjust input values and tolerances for each case instead of computing them.

Reply:
The values of `b` are modified so that the difference between `a` and `b` exceeds the tolerance.
I agree it's messy, I'll duplicate it instead.